### DISCUSSION TITLE
Why use squash merging?

### DISCUSSION THREAD

**Discussion by jcbhmr** [spenserblack/actions-wiki#43] [Q&A] [answered]
imo in the cases ive seen in this repo it's caused me more pain than it helps 🤷

_also its not the default and u know I'm a default lover_

but there might be a legit reason?

**Comment by spenserblack** [comment, marked as answer]
I prefer to avoid adding commits that get undone or are partial. It's just kind of extra noise in the commit history. For example, commits that exist just for formatting, or 2 or more commits that could be described as just one. Alternative to squash merges, I could ask you to interactively rebase your branch, but that could cause even more pain to you :laughing:

I don't always squash-merge, and try to take things on a case-by-case basis. But the majority of PRs can be summarized as one change, so I'll squash-merge those.

If you're curious, I wrote some of my opinions on git history here: https://spenserblack.github.io/tutorials/git

**Reply by jcbhmr** [reply]
i guess i do agree. my vehamant opposition was because I had started a pr that got completely out of sync because the squash commit _didnt align_ with the history 🤬

but if I had rebased against main instead of manual merge conflicts... hmmmm 🤔

I guess it was a combo of my bad and a weird edge case of merging magic. documentation in contributing.md about merge/squash would be a good idea lmao 😜

---

after googling around for other devs opinions (in addition to yours) i kinda like this quote

> I can honestly see the advantages and disadvantages of both sides of the debate. I don’t have a good answer for which should be the default option for a codebase. I like the ability to see every commit, but I also like a cleaner looking history. I think this puts me in a minority simply because I haven’t become enthusiastically attached to one way or the other.

&mdash; https://www.lloydatkinson.net/posts/2022/should-you-squash-merge-or-merge-commit/

squash is, like you said, awesome at condensing simple back-and-forth changes into one, while regular merge is useful for larger long-lived items where there might be **multiple authors** and **multiple areas** that were touched that you want to stay separate commits
